x-pack/plugins/security/server/authentication/authenticator.ts

```diff
@@ -349,27 +349,33 @@ export class Authenticator {
     assertRequest(request);
 
     const existingSessionValue = await this.getSessionValue(request);
-    const suggestedProviderName =
-      existingSessionValue?.provider.name ??
-      request.url.searchParams.get(AUTH_PROVIDER_HINT_QUERY_STRING_PARAMETER);
-
     if (this.shouldRedirectToLoginSelector(request, existingSessionValue)) {
-      this.logger.debug('Redirecting request to Login Selector.');
+      const providerNameSuggestedByHint = request.url.searchParams.get(
```

Comment:
__note:__ `authenticate` is invoked for every request, so I tried to call `request.url.searchParams.get` only when it's really needed (and that's why I have `searchParams.get` here and below).